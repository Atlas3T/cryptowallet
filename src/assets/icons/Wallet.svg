<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 22.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 51 51" style="enable-background:new 0 0 51 51;" xml:space="preserve">
<style type="text/css">
	.st0{fill:none;stroke:#000000;stroke-width:5;stroke-miterlimit:10;}
	.st1{display:none;}
	.st2{display:inline;fill:none;stroke:#000000;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:10;}
	.st3{display:inline;}
</style>
<g id="Wallet">
	<path class="st0" d="M46.1,31.5v5.3c0,2.4-1.9,4.3-4.3,4.3H9.5c-2.4,0-4.3-1.9-4.3-4.3V15.5c0-2.4,1.9-4.3,4.3-4.3h32.3
		c2.4,0,4.3,1.9,4.3,4.3v5.2"/>
	<circle cx="46.1" cy="26.1" r="3.1"/>
</g>
<g id="Payments" class="st1">
	<line class="st2" x1="8" y1="19.8" x2="41.8" y2="19.8"/>
	<polyline class="st2" points="35.8,13.3 42.3,19.8 35.8,26.3 	"/>
	<line class="st2" x1="42.3" y1="32.8" x2="8.5" y2="32.8"/>
	<polyline class="st2" points="14.5,39.3 8,32.8 14.5,26.3 	"/>
</g>
<g id="Exchange" class="st1">
	<path class="st2" d="M15.3,21.7c0-5.6,4.6-10.2,10.2-10.2s10.2,4.6,10.2,10.2"/>
	<polyline class="st2" points="42.2,16.3 35.7,22.8 29.2,16.3 	"/>
	<path class="st2" d="M35.7,30.2c0,5.6-4.6,10.2-10.2,10.2s-10.2-4.6-10.2-10.2"/>
	<polyline class="st2" points="8.8,35.7 15.3,29.2 21.9,35.7 	"/>
</g>
<g id="More" class="st1">
	<circle class="st3" cx="26" cy="15.8" r="7.8"/>
	<circle class="st3" cx="12.3" cy="28.6" r="5"/>
	<circle class="st3" cx="35.3" cy="33.6" r="3.5"/>
</g>
</svg>
