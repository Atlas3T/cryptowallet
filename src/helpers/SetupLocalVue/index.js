import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import messages from '@/i18n/';
import axios from 'axios';
import { sync } from 'vuex-router-sync';
import errorHandlerPlugin from '@/boot/ErrorHandler';
import toasterPlugin from '@/boot/Toaster';
import permissionsPlugin from '@/boot/Permissions';
import appInvitationPlugin from '@/boot/AppInvitation';
import smsPlugin from '@/boot/Sms';
import emailPlugin from '@/boot/Email';
import contactsImportPlugin from '@/boot/ContactsImport';
import * as All from 'quasar'

const { Quasar, date } = All;

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})


const localVue = createLocalVue();

localVue.use(Quasar, { components });
localVue.use(VueI18n);
localVue.use(Vuex);
// localVue.use(VueRouter);
localVue.prototype.$axios = axios;
errorHandlerPlugin({ Vue: localVue });
toasterPlugin({ Vue: localVue });
permissionsPlugin({ Vue: localVue });
appInvitationPlugin({ Vue: localVue });
smsPlugin({ Vue: localVue });
emailPlugin({ Vue: localVue });
contactsImportPlugin({ Vue: localVue });

/**
 * creates a new router and syncs it with the store using vuex-router-sync
 * @param store
 * @returns router
 */
const createRouter = (store = null) => {
  const router = new VueRouter({
    routes: [
      {
        path: '/contact/:id',
      },
    ],
  });

  if (store) { sync(store, router); }
  return router;
};

const i18n = new VueI18n({ messages, locale: 'en-gb' });

export {
  localVue,
  createRouter,
  i18n,
};
