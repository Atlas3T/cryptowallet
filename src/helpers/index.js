import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
