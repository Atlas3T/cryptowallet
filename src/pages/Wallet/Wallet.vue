<template>
  <div class="layout">
    <div class="list">
      <WalletsList
        :coins="coins"
        :currency="currency"
      />
    </div>
    <div class="button">
      <AddWallet/>
    </div>
  </div>
</template>

<script>
import WalletsList from '@/components/Wallet/Wallet/WalletsList.vue';
import AddWallet from '@/components/Wallet/Wallet/AddWallet.vue';

export default {
  name: 'Balance',
  components: {
    WalletsList,
    AddWallet,
  },
  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      currency: 'GPB',
      coins: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
