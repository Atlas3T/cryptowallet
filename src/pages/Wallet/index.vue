<template>
  <div class="layout">
    <div class="list">
      <wallet-list
        :wallets="wallets"
        :currency="selectedCurrency"
      />
    </div>
    <div class="button">
      <add-wallet/>
    </div>
  </div>
</template>

<script>
import WalletList from '@/components/Wallet/WalletsList';
import AddWallet from '@/components/Wallet/AddWallet';

export default {
  name: 'Balance',

  components: {
    WalletList,
    AddWallet,
  },

  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      wallets: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },

  computed: {
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
