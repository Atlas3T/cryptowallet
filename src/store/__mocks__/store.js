import Vue from 'vue';
import Vuex from 'vuex';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from '@atlascity/vuex-orm-plugin-lokijs';

import wallet from '@/store/wallet';
import contacts from '@/store/contacts/__mocks__/contacts.js';
import search from '@/store/search/__mocks__/search.js';
import settings from '@/store/settings/__mocks__/settings.js';
import setup from '@/store/setup/__mocks__/setup.js';
import qrcode from '@/store/qrcode/__mocks__/qrcode.js';
import modals from '@/store/modals/__mocks__/modals.js';

// entities
import Account from '@/store/wallet/entities/account';
import Address from '@/store/wallet/entities/address';
import Tx from '@/store/wallet/entities/tx';
import Utxo from '@/store/wallet/entities/utxo';
import Wallet from '@/store/wallet/entities/wallet';
import Coin from '@/store/wallet/entities/coin';
import KeyPair from '@/store/wallet/entities/keyPair';
import Prices from '@/store/prices';
import LatestPrice from '@/store/latestPrice';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
  Wallet,
];

const options = {
  env: 'browser',
};

VuexORM.use(VuexORMLoki, { database, options });

export const actions = {};
export const getters = {};
export const mutations = {};
export const state = {};

export function createMocks(custom = {
  getters: {}, mutations: {}, actions: {}, state: {},
}) {
  /**
   * mock store modules.
   */
  const modules = {
    wallet,
    search,
    settings,
    contacts,
    setup,
    qrcode,
    modals,
  };

  let mockGettersCombined = {};
  let mockMutationsCombined = {};
  let mockActionsCombined = {};
  let mockStateCombined = {};

  Object.entries(modules).forEach(([name, module]) => {
    mockGettersCombined = Object.assign({}, mockGettersCombined, module.getters);
    mockMutationsCombined = Object.assign({}, mockMutationsCombined, module.mutations);
    mockActionsCombined = Object.assign({}, mockActionsCombined, module.actions);
    mockStateCombined = Object.assign({}, mockStateCombined, module.state);

    if (custom.state && custom.state[name]) {
      module.state = Object.assign(module.state, custom.state[name]);
    }

    modules[name] = {
      namespaced: true,
      getters: Object.assign({}, getters, module.getters),
      mutations: Object.assign({}, mutations, module.mutations),
      actions: Object.assign({}, actions, module.actions),
      state: Object.assign({}, state, module.state),
    };
  });

  mockGettersCombined = Object.assign({}, mockGettersCombined, custom.getters);
  mockMutationsCombined = Object.assign({}, mockMutationsCombined, custom.mutations);
  mockActionsCombined = Object.assign({}, mockActionsCombined, custom.actions);
  mockStateCombined = Object.assign({}, mockStateCombined, custom.state);
  return {
    getters: mockGettersCombined,
    mutations: mockMutationsCombined,
    actions: mockActionsCombined,
    state: mockStateCombined,
    store: new Vuex.Store({
      getters: mockGettersCombined,
      mutations: mockMutationsCombined,
      actions: mockActionsCombined,
      state: mockStateCombined,
      modules,
      plugins: [VuexORM.install(database)],
    }),
  };
}

export const { store } = createMocks();
