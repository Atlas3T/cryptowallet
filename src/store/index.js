/*eslint-disable*/

import Vue from 'vue';
import Vuex from 'vuex';
import VuexPersistence from 'vuex-persist';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from 'vuex-orm-lokijs';

// import entities.
import Account from './wallet/entities/account';
import Address from './wallet/entities/address';
import Tx from './wallet/entities/tx';
import Utxo from './wallet/entities/utxo';
import Wallet from './wallet/entities/wallet';
import Coin from './wallet/entities/coin';
import KeyPair from './wallet/entities/keyPair';
import Prices from './prices';
import LatestPrice from './latestPrice';

// import modules.
import payments from './payments';
import search from './search';
import settings from './settings';
import setup from './setup';
import contacts from './contacts';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
];

window.Tx = Tx;
window.Utxo = Utxo;
window.Address = Address;
window.Wallet = Wallet;
window.LatestPrice = LatestPrice;
window.Prices = Prices;
window.Account = Account;
window.Coin = Coin;

// Setup persistant storage.
const vuexLocal = new VuexPersistence({
  supportCircular: true,
  storage: window.localStorage,
  modules: [settings],
});


const options = {
  env: 'browser',
};

function hydrationCompletedCallback() {
  setTimeout(() => {
    store.dispatch('settings/setLoading', false);
  }, 1000);
}

VuexORM.use(VuexORMLoki, { database, options, hydrationCompletedCallback });


/**
 * Create CryptoWallet Vuex store obj.
 * @type {Store<any>}
 */
const store = new Vuex.Store({
  modules: {
    payments,
    search,
    settings,
    setup,
    contacts,
  },
  plugins: [VuexORM.install(database)],
});

Vue.prototype.activeWallets = {};

// @todo remove line below (used for development purposes only)
window.store = store;

export default store;
