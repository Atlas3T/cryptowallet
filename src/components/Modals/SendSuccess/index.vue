<template>
  <div>
    <q-modal
      v-model="sendSuccessModalOpened"
      class="dark-modal modal"
    >
      <div class="header-section">
        <div class="header-back-button-wrapper">
          <q-btn
            icon="arrow_back"
            size="lg"
            class="icon-btn back-arrow-btn"
            flat
            @click.prevent="complete"
          />
        </div>
        <h1 class="header-h1">Success</h1>
      </div>
      <div
        v-if="wallet && txData"
        class="modal-layout-wrapper center-modal-content"
      >
        <div class="done-content-wrapper">
          <div class="done-icon-wrapper">
            <q-icon
              name="done"
              class="done-icon"
            />
          </div>

          <div class="done-msg-wrapper">
            <h1>PLAYFUL SUCCESS MESSAGE HERE</h1>
          </div>

          <div class="done-tx-details-wrapper">
            <div class="tx-details-row">
              <div class="detail">Tx hash</div>
              <div class="break">
                <span class="tx-hash-span">{{ txData.transaction.hash }}</span>
                <q-btn
                  :label="$t('copy')"
                  color="blueish"
                  size="xs"
                  class="copy-hash-btn"
                  @click="copyToClipboard"
                />
              </div>
            </div>

            <div class="tx-details-row">
              <div class="detail">Recipient</div>
              <div class="break">{{ recipient }}</div>
            </div>

            <div class="tx-details-row">
              <div class="detail">Amount</div>
              <div class="nowrap">{{ amount }}</div>
            </div>
          </div>

          <div class="send">
            <q-btn
              :label="$t('close')"
              color="blueish"
              size="md"
              @click="complete"
            />
          </div>
        </div>
      </div>
    </q-modal>
  </div>
</template>

<script>
import { mapState } from 'vuex';
import { AmountFormatter } from '@/helpers';
import Coin from '@/store/wallet/entities/coin';

export default {
  name: 'SendSuccess',
  data() {
    return {
      sendSuccessModalOpened: false,
      txData: null,
    };
  },
  computed: {
    ...mapState({
      id: (state) => { return state.route.params.id; },
      authenticatedAccount: (state) => { return state.settings.authenticatedAccount; },
    }),
    wallet() {
      return this.$store.getters['entities/wallet/find'](this.id);
    },
    supportedCoins() {
      return Coin.all();
    },
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
    coinSymbol() {
      return this.supportedCoins.find((coin) => { return coin.name === this.wallet.name; }).symbol;
    },
    amount() {
      const inCurrency = this.coinToCurrency(this.txData.transaction.value);
      return `${this.txData.transaction.value} ${this.coinSymbol} (${inCurrency})`;
    },
    recipient() {
      if (Array.isArray(this.txData.transaction.receiver)) {
        return this.txData.transaction.receiver[0];
      }
      return this.txData.transaction.receiver;
    },
    latestPrice() {
      const prices = this.$store.getters['entities/latestPrice/find'](`${this.coinSymbol}_${this.selectedCurrency.code}`);
      return prices.data.PRICE;
    },
  },
  mounted() {
    this.$root.$on('sendSuccessModalOpened', (value, txData) => {
      this.sendSuccessModalOpened = value;
      this.txData = txData;
    });
  },
  methods: {
    complete() {
      this.sendSuccessModalOpened = false;
      this.$router.go(-1);
    },
    coinToCurrency(amount) {
      const formattedAmount = new AmountFormatter({
        amount,
        rate: this.latestPrice,
        format: '0,0[.]00',
        coin: this.wallet.name,
        prependPlusOrMinus: false,
        currency: this.selectedCurrency,
        toCurrency: true,
        withCurrencySymbol: true,
      });

      return formattedAmount.getFormatted();
    },
    copyToClipboard() {
      try {
        cordova.plugins.clipboard.copy(this.address);
        this.$toast.create(0, this.$t('copied'), 200);
      } catch (err) {
        this.errorHandler(err);
      }
    },
  },
};
</script>

<style scoped>
.small-text {
  font-size: 0.8rem;
}

.sending-spinner-overlay {
  position: absolute;
  height: 100vh;
  width: 100vw;
  z-index: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgba(255,255,255,0.9);
  opacity: 0;
  transition: all ease-in-out 250ms;
}

.sending-spinner-overlay.active{
  opacity: 1;
  z-index: 2;
}

.done-content-wrapper,
.center-modal-content {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;
}

.done-icon-wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
}

.done-icon-wrapper {
  border: 5px solid #8eff8e;
  height: 7rem;
  width: 7rem;
  border-radius: 100%;
  text-shadow: 1px 1px 0px rgba(0, 0, 0, 0.5);
  color: #8eff8e;
}

.done-icon {
  font-size: 5rem;
  font-weight: bold;
}

.done-msg-wrapper {
  text-align: center;
  font-family: 'CooperHewitt-SemiboldItalic';
  text-shadow: 1px 1px 0px rgba(0, 0, 0, 0.5);
}

.done-msg-wrapper h1 {
  margin-top: 1.5rem;
}

.done-tx-details-wrapper {
  width: 100%;
  font-size: 0.7rem;
  background: #17334c;
  margin-top: 1rem;
  border-radius: 0.5rem;
}

.tx-details-row {
  display: flex;
  border-bottom: 1px solid #2f465a;
  padding: 0.7rem 1rem;
  line-height: 1.5;
}

.tx-details-row:last-of-type {
  border-bottom: none;
}

.break {
  word-break: break-all;
}

.nowrap {
  white-space: nowrap;
}

.detail {
  width: 12rem;
}

.copy-hash-btn {
  font-size: 8px;
  padding: 0.1rem 0.5rem;
  height: 0.9rem;
  line-height: 0;
  min-height: auto;
  position: relative;
  top: -1px;
}

.tx-hash-span {
  margin-right: 0.5rem;
}
</style>
