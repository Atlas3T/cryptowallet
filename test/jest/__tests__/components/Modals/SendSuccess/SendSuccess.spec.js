/* eslint-disable no-magic-numbers */
import SendSuccess from '@/components/Modals/SendSuccess/SendSuccessContent.vue';
import { mount } from '@vue/test-utils';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';
import Wallet from '@/store/wallet/entities/wallet';
import cordovaMocks from '~/test/CordovaMocks';
import Coin from '@/store/wallet/entities/coin';
import LatestPrice from '@/store/latestPrice';

describe('modals/SendSuccess', () => {
  let wrapper;
  let router;
  let storeMocks;

  const walletData = JSON.parse('{"$id":3,"id":3,"account_id":1,"name":"Bitcoin","displayName":"Bitcoin","symbol":"BTC","sdk":"Bitcoin","network":"BITCOIN_TESTNET","internalChainAddressIndex":23,"externalChainAddressIndex":15,"externalAddress":"2NGBz7mknbB1GxFSddxa47C3S6qS4FuTnyd","confirmedBalance":0.52543197,"unconfirmedBalance":0,"imported":false,"enabled":false,"lastBlockHeight":0,"parentName":"","parentSdk":"","contractAddress":"","decimals":"","hdWallet":{"bip":49,"ext":{"xpriv":"xprv9ztbaJYMADTpxd5o1wyCqJAfnBjbLvSqWsqgk2HKSgDsi4BQPBf5PdJ4XLPZd4ugHQ9o45sExCuJuWBnxyu61Gt7fVj1kAPoB1hsze1XNAC","xpub":"xpub6Dswyp5Ezb28B7AG7yWDCS7QLDa5kPAgt6mHYQgw11krarWYviyKwRcYNbeutQRsNtQn1DD9SnaXHMpwpzAEDnBxvZZrDFQ2ehUGw8kxH6u"},"int":{"xpriv":"xprv9ztbaJYMADTpzuEkRPVdJkHNUmdBg98S2VUv91asoQzVTZtHsavn9SYuxy4iYwjpLZrmNy4oUJQoS6fGUJBMvT7Kfm3uvw1Q9b3DhJY23i5","xpub":"xpub6Dswyp5Ezb28DPKDXR2dftE72oTg5brHPiQWwPzVMkXULNDSR8F2hEsPpEZKhymsboomxhqiwmnvec5FTc36WuFWSGPtjQrL6YrQ6JWSBnh"},"type":1,"network":{"name":"BITCOIN_TESTNET","type":"testnet","bip":1,"segwit":true,"discovery":"http://92.207.178.198:3001/api","broadcastUrl":"https://chain.so/api/v2/send_tx/BTCTEST","feeApi":"https://api.blockcypher.com/v1/btc/main","connect":{"messagePrefix":"0018Bitcoin Signed Message:","bech32":"tb","bip32":{"public":70617039,"private":70615956},"pubKeyHash":111,"scriptHash":196,"wif":239}}},"erc20Wallet":""}');

  const bitcoinCoinData = JSON.parse('{"$id":"Bitcoin","name":"Bitcoin","displayName":"Bitcoin","minConfirmations":6,"sdk":"Bitcoin","symbol":"BTC","network":"BITCOIN_TESTNET","denomination":"0.00000000","parentName":"","parentSdk":"","contractAddress":"","decimals":""}');
  const latestPriceData = JSON.parse('{"$id":"BTC_GBP","coin":"BTC","currency":"GBP","updated":1554305869988,"data":{"VOLUME24HOURTO":10147125.523436274,"PRICE":3818.12,"CHANGEPCT24HOUR":4.734003741558176}}');

  const customStore = {
    state: {
      modals: {
        sendConfirmTxData: {
          transaction: {
            receiver: ['123'],
            value: 1,
            fee: 0.01,
            hash: '0000',
          },
        },
      },
    },
  };

  const defaultProps = {};
  function wrapperInit(options) {
    return mount(SendSuccess, options);
  }

  function storeInit(custom, propsData, path = '/wallet/single/send/3') {
    storeMocks = createStoreMocks(custom);
    Wallet.insert({ data: [walletData] });
    LatestPrice.insert({ data: latestPriceData });
    Coin.insert({ data: bitcoinCoinData });

    router = createRouter(storeMocks.store);
    router.push({ path });
    wrapper = wrapperInit({
      i18n,
      router,
      localVue,
      propsData,
      store: storeMocks.store,
      mocks: {
        errorHandler: jest.fn(),
      },
    });
  }

  beforeEach(() => {
    cordovaMocks.initMocks();
    storeInit(customStore, defaultProps);
  });

  afterEach(() => { cordovaMocks.destroyMocks(); });

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the txData', () => {
    expect(wrapper.text()).toMatch('1 BTC');
    expect(wrapper.text()).toMatch('0000');
    expect(wrapper.text()).toMatch('123');
  });

  it('closes the modal and returns to previous page when close button is clicked ', () => {
    wrapper.findAll('button').at(2).trigger('click');
    expect(storeMocks.actions.setSendSuccessModalOpened.mock.calls[0][1]).toBe(false);
  });

  describe('copyToClipboard()', () => {
    it('can copy the address when the copy button is clicked', (done) => {
      wrapper.vm.$toast.create = jest.fn();
      cordova.plugins.clipboard.mockBehaviour = 1;
      wrapper.findAll('button').at(1).trigger('click');
      setTimeout(() => {
        expect(wrapper.vm.$toast.create).toHaveBeenCalled();
        done();
      }, 0);
    });

    it('can handle errors if tx cannot be copied', (done) => {
      customStore.state.modals.sendConfirmTxData.transaction.receiver = '123';
      storeInit(customStore, defaultProps);
      wrapper.vm.$toast.create = jest.fn();
      cordova.plugins.clipboard.mockBehaviour = 2;
      wrapper.findAll('button').at(1).trigger('click');
      setTimeout(() => {
        expect(wrapper.vm.errorHandler).toHaveBeenCalled();
        done();
      }, 0);
    });
  });
});
