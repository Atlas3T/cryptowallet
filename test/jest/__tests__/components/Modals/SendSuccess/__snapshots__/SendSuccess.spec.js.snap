// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modals/SendSuccess renders and matches snapshot 1`] = `
<div>
  <div
    class="header-section"
  >
    <div
      class="header-back-button-wrapper"
    >
      <button
        class="q-btn inline q-btn-item non-selectable icon-btn back-arrow-btn q-btn--flat q-btn--rectangle q-focusable q-hoverable"
        style="font-size: 20px;"
        tabindex="0"
        type="button"
      >
        <div
          class="q-focus-helper"
          tabindex="-1"
        />
        <div
          class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
        >
          <i
            aria-hidden="true"
            class="material-icons q-icon"
          >
            arrow_back
          </i>
        </div>
      </button>
    </div>
     
    <h1
      class="header-h1"
    >
      
      Success
    
    </h1>
  </div>
   
  <div
    class="modal-layout-wrapper center-modal-content"
  >
    <div
      class="done-content-wrapper"
    >
      <div
        class="done-icon-wrapper"
      >
        <i
          aria-hidden="true"
          class="material-icons q-icon done-icon"
        >
          done
        </i>
      </div>
       
      <div
        class="done-msg-wrapper"
      >
        <h1>
          PLAYFUL SUCCESS MESSAGE HERE
        </h1>
      </div>
       
      <div
        class="done-tx-details-wrapper"
      >
        <div
          class="tx-details-row"
        >
          <div
            class="detail"
          >
            
            TX Hash:
          
          </div>
           
          <div
            class="break"
          >
            <span
              class="tx-hash-span"
            >
              0000
            </span>
             
            <button
              class="q-btn inline q-btn-item non-selectable copy-hash-btn q-btn--standard q-btn--rectangle bg-blueish text-white q-focusable q-hoverable"
              style="font-size: 8px;"
              tabindex="0"
              type="button"
            >
              <div
                class="q-focus-helper"
                tabindex="-1"
              />
              <div
                class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
              >
                <div>
                  Copy
                </div>
              </div>
            </button>
          </div>
        </div>
         
        <div
          class="tx-details-row"
        >
          <div
            class="detail"
          >
            
            Recipient
          
          </div>
           
          <div
            class="break"
          >
            
            123
          
          </div>
        </div>
         
        <div
          class="tx-details-row"
        >
          <div
            class="detail"
          >
            
            Amount
          
          </div>
           
          <div
            class="nowrap"
          >
            
            1 BTC (Â£3,818.12)
          
          </div>
        </div>
      </div>
       
      <div
        class="send"
      >
        <button
          class="q-btn inline q-btn-item non-selectable q-btn--standard q-btn--rectangle bg-blueish text-white q-focusable q-hoverable"
          style="font-size: 14px;"
          tabindex="0"
          type="button"
        >
          <div
            class="q-focus-helper"
            tabindex="-1"
          />
          <div
            class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
          >
            <div>
              Close
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
